src/com/facebook/buck/ide/intellij/BaseIjModuleRule.java

```diff
@@ -100,14 +106,39 @@ protected void addSourceFolders(
     }
   }
 
-  private void addDepsAndFolder(
+  protected void addResourceFolders(
+      ResourceFolderFactory factory,
+      ImmutableMultimap<Path, Path> foldersToInputsIndex,
+      Path resourcesRoot,
+      ModuleBuildContext context) {
+    for (Map.Entry<Path, Collection<Path>> entry : foldersToInputsIndex.asMap().entrySet()) {
+      context.addSourceFolder(
+          factory.create(
+              entry.getKey(),
+              resourcesRoot,
+              ImmutableSortedSet.copyOf(Ordering.natural(), entry.getValue())));
+    }
+  }
+
+  private void addDepsAndFolderWithFolderInputIndexFilter(
```

Comment:
Better name would be `addDepsAndFoldersWithFilteringIfNeeded` or just `addDepsAndFoldersWithFiltering`